modules/dreamview/frontend/src/renderer/obstacles.js

```diff
@@ -128,17 +113,101 @@ export default class PerceptionObstacles {
                     y: position.y,
                     z: position.z + obstacle.height + 0.5,
                 };
-                this.updateIcon(iconPosition, world.autoDrivingCar.heading, iconIdx, scene);
-                iconIdx++;
+                this.updateIcon(iconPosition, world.autoDrivingCar.heading, this.iconIdx, scene);
+                this.iconIdx++;
             }
         }
-        hideArrayObjects(this.arrows, arrowIdx);
-        hideArrayObjects(this.solidCubes, cubeIdx);
-        hideArrayObjects(this.dashedCubes, cubeIdx);
-        hideArrayObjects(this.extrusionSolidFaces, extrusionFaceIdx);
-        hideArrayObjects(this.extrusionDashedFaces, extrusionFaceIdx);
-        hideArrayObjects(this.icons, iconIdx);
-        hideArrayObjects(this.trafficCones, trafficConeIdx);
+        this.hideUnusedObjects();
+    }
+
+    updateSensorMeasurements(world, coordinates, scene) {
+        if (!STORE.options['showObstaclesLidarSensor'] && !STORE.options['showObstaclesRadarSensor']
+            && !STORE.options['showObstaclesCameraSensor']) {
+            return;
+        }
+
+        const sensorMeasures = world.sensorMeasurements;
+        this.arrowIdx = this.arrows.length;
+        this.trafficConeIdx = this.trafficCones.length;
+        this.cubeIdx = this.solidCubes.length;
+        this.extrusionFaceIdx = this.extrusionSolidFaces.length;
+        this.iconIdx = this.icons.length;
+        for (const key in sensorMeasures) {
+            const sensorType = this.deduceSensorType(key.toLowerCase());
+            if (!sensorType || !STORE.options['showObstacles' + sensorType]) {
+                continue;
+            }
+
+            for (const measurement of sensorMeasures[key]["sensorMeasurement"]) {
+                if (!STORE.options['showObstacles' + _.upperFirst(_.camelCase(measurement.type))]
```

Comment:
let's remove the "showObstaclesXXX" check in this if statement. It would be better to show everything that this sensor type detects for better debugging purpose.